test/irc/test_irc_abstract_backends.py

```diff
@@ -298,3 +300,31 @@ def test_send_notice_safe():
     expected = 'NOTICE #sopel :Helloworld!\r\n'
     assert backend.message_sent == [expected.encode('utf-8')]
     assert bot.message_sent == [expected]
+
+
+def test_decode_line_utf8():
+    bot = BotCollector()
+    backend = MockIRCBackend(bot)
+    backend.isupport = ISupport()
+
+    test = "PRIVMSG #sopel :Hello, Martín!"
+    assert backend.decode_line(test.encode("utf-8")) == test
+
+
+def test_decode_line_utf8only():
+    bot = BotCollector()
+    backend = MockIRCBackend(bot)
+    backend.isupport = ISupport(UTF8ONLY=None)
+
+    test = "PRIVMSG #sopel :Hello, Martín!"
+    with raises(UnicodeDecodeError):
```

Comment:
I tend to prefer `with pytest.raises` because it's clearer. Others (Exi) might disagree, but I like to keep the namespace of imported stuff. See-also: My preference for e.g. `from sopel import plugin` and then `@plugin.command()`.

(Forgot to add this _before_ submitting that review just now, sorry.)

Reply:
Yep, fair. It's also tests so being thrifty with includes isn't exactly priority zero.